src/Avalonia.Controls/VirtualizingStackPanel.cs

```diff
@@ -51,7 +52,25 @@ public class VirtualizingStackPanel : VirtualizingPanel, IScrollSnapPointsInfo
             RoutedEvent.Register<VirtualizingStackPanel, RoutedEventArgs>(
                 nameof(VerticalSnapPointsChanged),
                 RoutingStrategies.Bubble);
+        /// <summary>
+        /// Defines the <see cref="BufferFactor"/> property.
+        /// </summary>
+        public static readonly StyledProperty<double> BufferFactorProperty =
+            AvaloniaProperty.Register<VirtualizingStackPanel, double>(nameof(BufferFactor), 0.0, 
+                validate: v => v is >= 0 and <= 2);
 
+        /// <summary>
+        /// Gets or sets the factor to determine how much additional space to maintain above and below the viewport.
+        /// A value of 0.5 means half the viewport size will be buffered on each side (up-down or left-right)
+        /// This uses more memory as more UI elements are realized, but greatly reduces the number of Measure-Arrange
+        /// cycles which can cause heavy GC pressure depending on the complexity of the item layouts.
+        /// </summary>
+        public double BufferFactor
```

Comment:
Nit: Put properties with other properties (not at the top of the file with static fields)